### ISSUE TITLE
Can you provide a test demo

### ISSUE THREAD

**Issue by bleakie** [genbing99/SoftNet-SpotME#1]
Great work, is it convenient to provide a test demo？

**Comment by genbing99** [comment]
Hi bleakie,

You can view the web demo video at this link:
(hidden)

Note that this is just a demo to demonstrate how the spotting works in the long video, and this is not hosted on any website due to the huge storage and memory required to process.

**Comment by bleakie** [comment]
What I mean is how to use the model to judge macro expression and micro expression on a single image.

**Comment by genbing99** [comment]
The model is trained and tested on micro- and macro-expression separately.

**Comment by bleakie** [comment]
Do you have a corresponding paper?

**Comment by genbing99** [comment]
The paper is still currently under review, I will upload it once it passed the review stage.

**Comment by bleakie** [comment]
[result](https://i.loli.net/2021/04/09/8djxXLbDVW6uwUt.png)
Is red the micro expression  ground truth?

**Comment by genbing99** [comment]
Yes, the two consecutive red lines indicate a ground truth interval.

**Comment by bleakie** [comment]
In feature extraction, only micro expression information is extracted. Why not extract non micro expression features as negative samples？

**Comment by genbing99** [comment]
The non micro-expression features are extracted as well.
In the SoftNet training section, a pseudo-labeling function is used to label the frames into 1 or 0.

**Comment by bleakie** [comment]
Dear author, I have a few questions：
    

1. why remove global head movement by minus nose region
2. why Eye masking
3. ROI Selection -> Image resampling into 42x22x3

**Comment by genbing99** [comment]
Thank you for asking,

1. Nose region is the most rigid part of the facial region, reduction of optical flow features can eliminate the global movement. (Motivated by paper 'Spatio-temporal fusion for macro-and micro-expression spotting in long video sequences')
2. Eye masking can remove the influence of eye blinking. (Refer to paper 'Automatic micro-expression recognition from long video using a single spotted apex')
3. Image resampling can reduce the image size, thus reduce the computational cost while training the model.

**Comment by bleakie** [comment]
Thank you for your reply.
In addition, the DualTVL1OpticalFlow_create method is time-consuming. Why did you choose this method at that time?

**Comment by genbing99** [comment]
Dense optical flow leads to better accuracy when estimating the flow vectors, TV-L1 is popular in the field of micro-expression to extract the optical flow due to the robustness towards the noise.

**Comment by bleakie** [comment]
Do you have any research on micro expression recognition？

**Comment by genbing99** [comment]
So far haven't research on the recognition task.

**Comment by genbing99** [comment]
So far haven't done any research on the recognition task.

**Comment by bleakie** [comment]
I have some questions about CAS (me) 2, such as s33_dispust2,  it's obvious that 300-330,430-450 has an expression . Why doesn't it have an expression on the label?

**Comment by genbing99** [comment]
I am not so sure about the expression labels, maybe you can email the authors of CAS(ME)2 dataset for the details.

**Comment by bleakie** [comment]
Can you share this file (CASME_sq code_final.xlsx) with me? Thank you very much.

**Comment by bleakie** [comment]
Why use the landmark of the first frame? Is the landmark of the frame behind not right
https://github.com/genbing99/SoftNet-SpotME/blob/881eca7c5488df35d2fa8938c1dcf7afff19d91e/extraction_preprocess.py#L29

**Comment by genbing99** [comment]
I am not authorized to share the file without the consent of the authors, so sorry about this.
The reason to use the landmark locations of the first frame is to ensure the optical flow algorithm is able to track the movement of the face.

**Comment by bleakie** [comment]
![image](https://user-images.githubusercontent.com/23338843/122870406-37b71a80-d360-11eb-9fb0-a422e27eea9d.png)
Why is resnet18 better than resnet50?

**Comment by genbing99** [comment]
From our findings, the deeper/complex network might cause the overfitting for learning the features. Hence, we designed a shallow network.
